### DISCUSSION TITLE
Network shares - how to list them?

### DISCUSSION THREAD

**Discussion by pskowronek** [mucommander/mucommander#920] [General]
I'm on macOS and I'm trying to open `Network shares` in muC - I'm being asked for credentials, the IP/server is set to last value from clicking at `SMB...`...

So, how this `Network shares` suppose to work? I was thinking that it will show all available smb servers in my local network...... but I guess it is not. Any way to list such shares w/o specifying their IPs or names?

**Comment by meny** [comment]
+1

And also an issue in a similar topic. The network shares must be first clicked and mounted in the OS and only then are available in mucommander. I tried adding these as SMB, etc., but nothing works. The only solution is to mount these manually every day (they disconnect daily; corporate environment) and only then access them via bookmarks.

**Reply by ahadas** [reply]
@meny that's weird, I can connect to SMB shares without mounting them in the OS by specifying their URL, e.g., `smb://<url>/<share>`

**Reply by meny** [reply]
It is, but indeed the same link which works via Cmd+K in MacOS Finder, when copy-pasted into the SMB path in muCommander, will not work. I also tried many variations splitting the link across the various fields in muCommanders SMB dialog and entering credentials, but no luck. All paths in my bookmarks show with the red no-entry icon and I must reconnect them via Finder first.

**Comment by ahadas** [comment]
@pskowronek that's funny, I was wondering about this myself this week. I worked on adding ADB (#919) and I saw that in trolCommander they've added a menu that lists the devices, so I thought about adding an OSGi service that returns a menu from the plugin that is added to the drive buttons - and I saw we have those network shares and bonjour there. I was trying to see if I can extract them to be a different OSGi bundles that return such a service and realized that `smb://` is queried and wondered if it ever returned anything (btw, we had an issue before that `smb://<URL>/` didn't return the list of shares on the server - that is supposed to be fixed already). so long story short, I don't know if it ever worked but it also didn't work for me. and I ended up extracting the bonjour menu :)

**Reply by pskowronek** [reply]
hmm, but shouldn't be a way to simple go to 'network shares' and see all the smb servers out there in LAN?

**Reply by ahadas** [reply]
need to check this functionality in [jsch-ng](https://github.com/AgNO3/jcifs-ng)

**Reply by pskowronek** [reply]
probably related https://github.com/AgNO3/jcifs-ng/issues/22#issuecomment-352194573 - uPnP?

**Reply by ahadas** [reply]
> probably related [AgNO3/jcifs-ng#22 (comment)](https://github.com/AgNO3/jcifs-ng/issues/22#issuecomment-352194573) - uPnP?

yeah

**Comment by maiemi** [comment]
here, smb connections are working fine, but if I want to connect to a webdav, this will not work. neither in http nor in https. I first have to connect to the webdav via finder on my mac, then I can connect to the webdav in muCommander by choosing it under "volumes".  When I try to connect to the webdav directly with muCommander, I get the message "Dieser Ordner existiert nicht oder ist nicht verfügbar" (similar in English: "This target doesn't exist or is unreachable"). The login credentials are correct.

**Reply by ahadas** [reply]
we didn't test webdav for quite a while, it could well be that it's broken

**Comment by maiemi** [comment]
Hi there, it's me again :)
I am using a Macbook Pro with macOS 15.1. (Sequoia) and M1 Pro processor.
Today I tried to connect to a smb share with muCommander (the "connect to server" icon in the menu bar) but it fails.
The credentials are correct but the share is not connected. Instead of that, a message appears. I added the message  at this post.
If I connect the same share with the same credentials in finder, it works. After that, I can reach the share under "/Volumes/sharename" in muCommander.
The behavior is the same as written by many at 2003, May, 8th above.
<img width="621" alt="Bildschirmfoto 2024-11-06 um 12 22 06" src="https://github.com/user-attachments/assets/fba2f068-6c17-49ac-bce1-3f3990b4e583">

**Reply by maiemi** [reply]
This behavior was not on my Macbook Pro with INTEL processor, Here on my Macbook Pro with Apple Silicon processor the behavior is present.

**Comment by sebthemagnificient** [comment]
Problem still exists here on Windows 10 Pro - many shares accessible from the original explorer ( guest workgroup and credential domained) but only one ( a domain share ) is accessible by control + k → smb.
Workgroup completely inaccessible, either by the control + k creation or network shares button

**Comment by maiemi** [comment]
is there any news or a way to fix this behavior?

**Reply by meny** [reply]
No news on my side. I currently click on the share Alias inconveniently taking up space and visually disturbing me on the desktop every day to mount the network share, then close the window and continue with muCommander.

**Comment by meny** [comment]
ChatGPT might have the actual answer as to why it's happening:

This is a **known issue with muCommander’s SMB support**. It doesn’t use macOS’s own SMB stack (like Finder does), but instead relies on **JCIFS** (Java CIFS client), which is outdated and only supports **SMB1 (NTLMv1)**.

Your symptoms (Finder works, muCommander fails with “incorrect credentials”) happen because:

* **Modern Windows Server / Active Directory** defaults to SMB2/3 and NTLMv2 or Kerberos.
* JCIFS in muCommander can’t negotiate those protocols.
* When you type in correct credentials, the AD server rejects them because the auth mechanism is too old.

### Options / Workarounds

1. **Force NTLMv1/SMB1 on the server side** (not recommended, insecure, usually disabled by admins).
2. **Try a JCIFS fork with SMB2/3 support**, e.g. [[SMBj](https://github.com/hierynomus/smbj)](https://github.com/hierynomus/smbj) — but muCommander doesn’t support it out of the box.
3. **Use a different file manager** on macOS that relies on the system’s SMB stack (like ForkLift, Commander One, Double Commander with GVFS). These will behave like Finder.
4. **Mount the share in Finder first** (`smb://domain;user@server/share`) and then access the mounted volume via muCommander (navigate to `/Volumes/...`). That way muCommander just sees it as a local path.

**Reply by ahadas** [reply]
one should always check the accuracy of answers by those chatbots :)
I'm saying this because @ShayArtzi generously contributed [an improvement for SMB support and now we default to smbj rather than jcifs](https://github.com/mucommander/mucommander/pull/1313). I don't know if that fixes the issue though, could you check this with [the nightly build](https://github.com/mucommander/mucommander/actions/runs/17719284801) (see the artifacts at the bottom of the page)?

**Reply by meny** [reply]
Thanks for the tip! It didn't work for me with the two shares I tested, but I'll stick with that nightly for now.
